Setup/UpgradeSchema.php

```diff
@@ -563,6 +563,12 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             $connection->createTable($table);
         }
 
+        if (version_compare($context->getVersion(), '1.11.3', '<')) {
+            $connection->dropTrigger('trg_catalog_product_index_price_after_delete');
+            $connection->dropTrigger('trg_catalog_product_index_price_after_insert');
+            $connection->dropTrigger('trg_catalog_product_index_price_after_update');
+        }
```

Comment:
Not sure if we should drop the trigger here in this case as there could be other subscriptions on these tables. It would be best to advise our customers to temporarily switch out their Indexer Mode from `Update on Schedule` to `Update on Save` for **algolia_products** and back to remove all triggers and add the ones they need.

Magento definitely did not make this easy but that would be the safest way.

Reply:
Hi @bsuravech, thanks for checking!

When preparing this PR, first thing I tired was to switch to update on save, then back to update by schedule mode. However, the triggers were not removed for me, which is why I had to make a migration. I'm fairly certain that Magento creates missing triggers automatically every time we run `bin/magento setup:upgrade` - this is part of [recurring setup](https://github.com/magento/magento2/blob/2.3.2/app/code/Magento/Indexer/Setup/RecurringData.php#L47).

I will try again within following days and report.

Reply:
Good day @bsuravech !

Like I promised, I tried again. Here's a video:

https://vimeo.com/368516103
Password: algolia

If we don't drop triggers, price index subscription remains even when we try switching indexer mode to Update on save and back to Update by schedule. I checked why:

[\Magento\Indexer\Controller\Adminhtml\Indexer\MassOnTheFly::execute](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Indexer/Controller/Adminhtml/Indexer/MassOnTheFly.php#L30)
[\Magento\Indexer\Model\Indexer::setScheduled](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Indexer/Model/Indexer.php#L293)
[\Magento\Framework\Mview\View::unsubscribe](https://github.com/magento/magento2/blob/2.3.3/lib/internal/Magento/Framework/Mview/View.php#L224)

The code iterates subscriptions, and removes them. However, our subscription **does not exist** any more in mview.xml, and so it's not removed from triggers.

If we drop triggers, they will be recreated on subsequent recurring data install:

[\Magento\Indexer\Setup\RecurringData::install](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Indexer/Setup/RecurringData.php#L52)

Demo video with dropping triggers:

https://vimeo.com/368524421
Password: algolia

Please review :)

Reply:
Ah, I see, the process workflow should be change to update on save, then install the upgrade/patch, and then switch back to update on schedule. 

Seeing as the subscription handles the creation / removal of these mview triggers, would it be better to replicate the old subscription to remove it using the subscription factory? See suggested. 

```suggestion
$subscriptionInstance = $this->subscriptionFactory->create(
    [
        'view' => $this->getMviewView()->load('algolia_products'),
        'tableName' => 'catalog_product_index_price',
        'columnName' => 'entity_id',
        'subscriptionModel' => SubscriptionFactory::INSTANCE_NAME,
    ]
);
$subscriptionInstance->remove();
```
getMviewView() would be an instance of \Magento\Framework\Mview\ViewInterface to load algolia_products View. 

I haven't tested this out myself but remove() method from Subscription would handle any linkedViews. I would feel more comfortable with this method.

Reply:
Good catch @bsuravech !

I updated the code and tested when updating existing subscriptions and when installing Algolia for the first time.

Reply:
Amazing @unicoder88 ! Thanks for your effort on this!